lib/spack/spack/util/url.py

```diff
@@ -365,20 +289,3 @@ def parse_git_url(url):
             raise ValueError("bad port in git url: %s" % url)
 
     return (scheme, user, hostname, port, path)
-
-
-def is_url_format(url):
-    return re.search(r"^(file://|http://|https://|ftp://|s3://|gs://|ssh://|git://|/)", url)
-
-
-def require_url_format(url):
```

Comment:
Are we still enforcing this transitively through `urllib`? I.e. `urllib` will throw if given anything but an expected url and the error message is clear, or does this refactor mean this is no longer necessary and we're just converting any potential non url strings into urls? I think it's the latter but it's not immediately obvious.

Reply:
My thinking was that the parser can't be a validator, cause what scheme is supported depends on the call site. For buildcache etc we still have some validation that at least errors when a url is passed to `--directory` and vice versa, although I find it a bit pedantic to be honest ;p